### DISCUSSION TITLE
Add wildcard support to custom domain

### DISCUSSION THREAD

**Discussion by paour** [LocalCan/LocalCanApp#4] [Feature requests]
Rather than go though the DNS setup and verification each time I want to add a subdomain, I would prefer a one-shot wildcard redirect `*.localcan.mydomain.com` to `hash.cname.localcan.dev`, and then be able to add a new domain like `app1.localcan.mydomain.net` as easily as I can with `Create on LocalCan`.

<img width="397" alt="SCR-20240802-niyw" src="https://github.com/user-attachments/assets/ae9e343b-d2ad-4b02-bf5d-61c18ecabdaa">

**Comment by Jaarson** [comment]
This is a very good idea — thanks for sharing! However, it does make issuing certificates harder, as issuing wildcard certificates requires a different approach ([DNS-01 challenge](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge)). Either API access to your DNS server would be needed, or you would need to provide the certificate  manually.

Thanks for sharing. I will think about how to make this happen.

**Reply by paour** [reply]
A wildcard cert does not necessarily follow from a wildcard DNS.

You could keep the current logic of validating each subdomain's cert with an HTTP challenge, even as I redirect all `*.localcan.mydomain.com` DNS to your cname.

It's what I do with my traefik server: direct all `*.mydomain.com` traffic to its IP address, but it still creates a different cert for each populated subdomain.
